akka-javasdk/src/main/java/akka/javasdk/agent/UserMessage.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.agent;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS)
```

Comment:
If we need to use Jackson type info we shouldn't use the concrete class but a short identifier